cumulus_library/study_parser.py

```diff
@@ -369,9 +371,17 @@ def _load_and_execute_builder(
         # execute, since the subclass would otherwise hang around.
         table_builder_class = table_builder_subclasses[0]
         table_builder = table_builder_class()
-        table_builder.execute_queries(
-            cursor, schema, verbose=verbose, drop_table=drop_table, parser=parser
-        )
+        if generate_sql:
+            table_builder.prepare_queries(cursor, schema, parser=parser)
+            table_builder.comment_queries(doc_str=doc_str)
+            table_builder.write_queries(
+                filename=f"{filename.split('.')[0]}.sql",
```

Comment:
nit: is there a more robust / pathlib-y way to get the basename? Like `basename` (didn't check if that exists) or something? Just thinking about filenames with periods in them like `hello.kitty.txt`

Reply:
Oh there is -  `Path.stem`, which, i get the name, but odd choice.

I could? rewrite the write_queries method to natively take a Path as a single arg and do that coercion here.